### DISCUSSION TITLE
스크린샷 모드에서

### DISCUSSION THREAD

**Discussion by peterkangdev** [naverz/zepeto-studio-kor#445] [5. Q&A]
스크린샷 예제를 이용하여 스크린샷을 만들려고 합니다. 
예제 프로젝트는 잘 작동합니다. 

다른 프로젝트를 생성하여 ZepetoScreenShotModule 을 넣고, doc 에 적힌대로 그대로 진행했습니다.
실행을 누르면 셀카 UI 가 나옵니다. 
잘 되나 생각되어, 셀카 버튼을 누르면 전혀 작동하지 않습니다. 
살펴 보니까, 
Unity 를 Run 하고 보면, 아래 부분이 disable 되어 있더군요.

<img width="570" alt="스크린샷 2022-03-01 오후 1 56 40" src="https://user-images.githubusercontent.com/30133892/156108228-306d1ceb-d86d-43dd-9424-7d0874d8180e.png">

따로 설정하는 것도 없고, 그냥 예제에 있던 모듈을 그대로 가져왔습니다.
그래서, 여러가지 살펴보던중 UIController.ts 에 Awake() 에서 

        this.screenShot = this.screenShotModeModule.GetComponent<ScreenShotController>();
        this.screenShotModeManager = this.screenShotModeModule.GetComponent<ScreenShotModeManager>();

이 부분이 작동하지 않는것을 알았습니다. 
this.screenShot, this.screenShotModeManager 로그를 찍어보면 항상 null 이 나옵니다. 
근데, null 이 나온다고, disable 되는 것도 이상하고....
관련된 Exception 을 발견했는데, 아래와 같았습니다.

<img width="1081" alt="스크린샷 2022-03-01 오후 1 56 57" src="https://user-images.githubusercontent.com/30133892/156108527-f8424652-a9bf-41d0-bdf2-7591a25cab08.png">

이 경로에 modular.js 는 보이지도 않습니다. 
제가 뭘 잘못한 걸까요?
아님 뭔가 문제가 있었던 걸까요?

World 버전은 최신 1.1.5 버전 입니다. 
나머지 설정은 정상적입니다.
doc 에 나온대로 몇번이고 확인했습니다.

**Comment by pumfit** [comment]
혹시 불러오신 다른 프로젝트에서 패키지를 import하시고 ZepetoScripts - ScreenShotScripts에 7개 스크립트 모두 들어가 있으신가요?
![image](https://user-images.githubusercontent.com/46295539/156110909-3eaf4009-b12a-4cc5-9112-b0454456bd97.png)

저도 다른 프로젝트로 모듈을 import하니 스크립트가 모두 불러와지지 않아 비슷한 에러가 나 파일 구조를 살펴보니 유니티 임포트 과정에서 몇개 스크립트가 빠졌었어서 해당 스크립트만 다시 넣어주었더니 정상 작동되었습니다!

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요. 위 방법으로도 해결되지 않았다면
탐색기에 들어가서 meta파일을 삭제한 후 다시 실행시키면 정상 동작 할 것입니다.
이후에 다른 프로젝트에 적용된 스크립트를 옮길때는 메타 파일은 제외하고 본 파일만 옮겨주세요~
